lib/utils/queryable.js

```diff
@@ -119,14 +147,27 @@ const setter = ({ data, key, value, force }) => {
   // ['foo', 'bar', 'baz'] -> { foo: { bar: { baz:  {} } }
   const keys = parseKeys(key)
   const setKeys = (_data, _key) => {
-    // handles array indexes, making sure the new array is created if
-    // missing and properly casting the index to a number
-    const maybeIndex = Number(_key)
-    if (!Number.isNaN(maybeIndex)) {
+    // handles array indexes, converting valid integers to numbers,
+    // note that occurences of Symbol(append) will throw,
+    // so we just ignore these for now
+    let maybeIndex = Number.NaN
+    try {
+      maybeIndex = Number(_key)
+    } catch (err) {}
+    if (!Number.isNaN(maybeIndex))
       _key = maybeIndex
-      if (!Object.keys(_data).length)
-        _data = []
-    }
+
+    // creates new array in case key is an index
+    // and the array obj is not yet defined
+    const keyIsAnArrayIndex = _key === maybeIndex || _key === _append
+    const dataHasNoItems = !Object.keys(_data).length
+    if (keyIsAnArrayIndex && dataHasNoItems)
+      _data = []
```

Comment:
what happens if i first use an array index, but then later use a property, indicating it was meant to be an object the whole time?

Reply:
you mean something like: `npm pkg set foo[0]=foo foo.bar=bar` ?

Reply:
yup, exactly

Reply:
that is not working as expected! 😅 I guess I know exactly what unit test I missed, although I don't think it's directly related to this specific PR it's def something we want to fix asap - I'll work on it 😊

Reply:
thanks again @ljharb, excellent catch 😄 that scenario is also handled now, let me know what you think!

Reply:
FYI:

```
$ npm pkg set foo[0]=foo foo.bar=bar
npm ERR! code EOVERRIDEVALUE
npm ERR! Property foo.bar already has a value in place.

npm ERR! A complete log of this run can be found in:
npm ERR!     $HOME/.npm/_logs/2021-07-13T20_17_50_539Z-debug.log

$ npm pkg set foo[0]=foo foo.bar=bar --force
npm WARN using --force Recommended protections disabled.

$ git diff
diff --git a/package.json b/package.json
index 38c21cf..fc9db2a 100644
--- a/package.json
+++ b/package.json
@@ -6,5 +6,8 @@
   },
   "dependencies": {
     "@ruyadorno/slow-lifecycle-script": "^1.0.3"
+  },
+  "foo": {
+    "bar": "bar"
   }
 }
```